src/operator/src/req_convert/insert/stmt_to_region.rs

```diff
@@ -237,9 +419,25 @@ fn sql_value_to_grpc_value(
         )
         .context(crate::error::SqlCommonSnafu)?
     };
+    validate(&value)?;
+    Ok(value)
+}
 
-    let grpc_value = value_to_grpc_value(value);
-    Ok(grpc_value)
+fn validate(value: &Value) -> Result<()> {
+    match value {
+        Value::Json(value) => {
+            // Json object will be stored as Arrow struct in parquet, and it has the restriction:
+            // "Parquet does not support writing empty structs".
+            ensure!(
+                !value.is_empty_object(),
+                InvalidInsertRequestSnafu {
+                    reason: "empty json object is not supported, consider adding a dummy field"
+                }
+            );
+            Ok(())
+        }
+        _ => Ok(()),
+    }
```

Comment:
can we assume empty object is null?

Reply:
It can be treated as a null value. But currently the null value insertion in json column has some problems, I'm going to fix them as well as this one here in the following PRs.